aux/src/TaggedFrameTensorSet.cxx

```diff
@@ -154,12 +155,50 @@ bool Aux::TaggedFrameTensorSet::operator()(const input_pointer& in, output_point
             itv->push_back(ITensor::pointer(sumt));
         }
 
+        // per tag cmm
+        for(auto label_cmm : in->masks()) {
+            auto label = label_cmm.first;
+            log->trace("masks: {}", label);
+            auto cmm = label_cmm.second;
+            size_t nranges = 0;
+            for(auto ch_bl : cmm) {
+                nranges += ch_bl.second.size();
+            }
+            SimpleTensor<double>* ranges = new SimpleTensor<double>({nranges, 2});
+            Eigen::Map<Eigen::ArrayXXd> ranges_arr((double*)ranges->data(), nranges, 2);
+            SimpleTensor<double>* channels = new SimpleTensor<double>({nranges});
+            Eigen::Map<Eigen::ArrayXd> channels_arr((double*)channels->data(), nranges);
+
+            size_t ind = 0;
+            for(auto ch_bl : cmm) {
+                auto ch = ch_bl.first;
+                auto bl = ch_bl.second;
+                for(auto b : bl) {
+                    ranges_arr(ind, 0) = b.first;
+                    ranges_arr(ind, 1) = b.second;
+                    channels_arr(ind) = ch;
+                    ++ind;
+                }
+            }
+
+            auto& ranges_md = ranges->metadata();
```

Comment:
I think this metadata schema is okay but could be better "normalized".  Anytime I find myself adding an "_" to a key or constructing a value with delimiters I know I'm creating something that isn't normalized.

The issue is that the receiving code will have to be somewhat intelligent to parse and interpret the `type` value as constructed here.  If we use well normalized metadata then all of the parsing is handled and the interpretation is minimized.

We may use deep structures here so I suggest a root on a single top-level key `cmm` to hold a structure with key `type` (value "range" or "channel") and key `label` with the label.  Ie, 

```c++
ranges_md["cmm"]["type"] = "range";
ranges_md["cmm"]["label"] = label;
```

(Eventually this data may be converted to HDF5 and we need an algorithm that flattens this structure as HFD5 attributes must be flat.  I may have added support for this already, if not, it's an easy fix.)